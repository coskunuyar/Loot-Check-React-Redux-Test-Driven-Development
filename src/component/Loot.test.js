import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {Loot} from './Loot';
import {shallow , configure , mount} from 'enzyme';

configure({adapter: new Adapter()});

describe('Loot',() => {
    const mockFetchbitcoin = jest.fn();
    const props = {balance: 10 , bitcoin: {} , fetchBitcoin: mockFetchbitcoin}
    let loot = shallow(<Loot {...props}/>);

    it('render properly',() => {
        expect(loot).toMatchSnapshot();
    });

    describe('When mounted', () => {
        beforeEach(() => {
            props.fetchBitcoin = mockFetchbitcoin;
            loot = mount(<Loot {...props}/>)
        });

        it('dispatches the fetchbitcoin method it receives from props', () => {
            expect(mockFetchbitcoin).toHaveBeenCalled();
        });
    });
});