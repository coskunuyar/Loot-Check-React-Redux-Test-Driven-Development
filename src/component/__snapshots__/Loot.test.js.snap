// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot render properly 1`] = `ShallowWrapper {}`;
